services/worker/src/worker/job_runners/split/descriptive_statistics.py

```diff
@@ -591,38 +626,40 @@ def transform(cls, example: dict[str, Any]) -> Any:
         raise NotImplementedError
 
     @classmethod
-    def _compute_statistics(
-        cls,
-        parquet_directory: Path,
-        column_name: str,
-        n_samples: int,
-    ) -> SupportedStatistics:
+    def compute_transformed_data(
+        cls, parquet_directory: Path, column_name: str, transform_func: Callable[[Any], Any]
+    ) -> list[Any]:
         parquet_files = list(parquet_directory.glob("*.parquet"))
         transformed_values = []
         for filename in parquet_files:
-            shard_items = pq.read_table(filename, columns=[column_name]).drop_null().to_pydict()[column_name]
-            shard_transformed_values = (
-                thread_map(
-                    cls.transform,
-                    shard_items,
-                    desc=f"Transforming values of {cls.__name__} {column_name} for {filename.name}",
-                    leave=False,
-                )
-                if shard_items
-                else []
+            shard_items = pq.read_table(filename, columns=[column_name]).to_pydict()[column_name]
+            shard_transformed_values = thread_map(
+                transform_func,
+                shard_items,
+                desc=f"Transforming values of {cls.__name__} {column_name} for {filename.name}",
+                leave=False,
             )
             transformed_values.extend(shard_transformed_values)
+        return transformed_values
 
-        if not transformed_values:
+    @classmethod
+    def _compute_statistics(
+        cls,
+        parquet_directory: Path,
+        column_name: str,
+        n_samples: int,
+    ) -> SupportedStatistics:
+        transformed_values = cls.compute_transformed_data(parquet_directory, column_name, cls.transform)
```

Comment:
> we're computing the transformed data twice, once to create the duckdb file, and once to compute the stats, right? Is it an issue, or is it a detail that does not affect the computation time too much?

ah you're right i haven't though about that the same job is done twice honestly. i'm not sure if that's an issue as these transformations are quite fast, except for maybe audio/image, which are also fast considering that the data size is small but not as fast as polars' native measures.
It's possible to make the stats step depended on `duckdb-index` to avoid this double compute but in this case the error with fts index for example would cause stats to fail too. I'd personally avoid it. wdyt @lhoestq @AndreaFrancis ?

> If I remember well, before, we depended on the duckdb index file to compute the stats, then we changed to rely on polars instead, is this right?

We did not depend on the duckdb index file, we queried parquet files directly with duckdb. Now we query them with polars.

Reply:
I think it's fine as it is for now